spring-boot-project/spring-boot-tools/spring-boot-loader-tools/src/main/java/org/springframework/boot/loader/tools/Repackager.java

```diff
@@ -168,7 +169,7 @@ public void repackage(File destination, Libraries libraries) throws IOException
 	 * @throws IOException if the file cannot be repackaged
 	 * @since 1.3.0
 	 */
-	public void repackage(File destination, Libraries libraries,
+	public synchronized void repackage(File destination, Libraries libraries,
```

Comment:
Can you expand how thread safety relates to the issue you are referring to?

Reply:
I disabled the check 'source.equals(destination)' and created a test with identicals source and destination that runs repackage twice to reproduce the issue but it didn't fail. I said to myself may be it passed because my test is sequential!